### DISCUSSION TITLE
No assets when using DbtCliResource within an op to run a snapshot job

### DISCUSSION THREAD

**Discussion by freddyosprey** [dagster-io/dagster#18672] [Q&A] [answered]
I am currently trying to create a job that runs only dbt snapshots and I think I am misunderstanding DbtCliResource as I see the following when trying to run locally, i.e. there are no assets to materialise.

Any help would be much appreciated!
![Screenshot 2023-12-12 at 17 49 08](https://github.com/dagster-io/dagster/assets/153658142/4b20e4eb-a0af-4ef9-8d20-f986f445cc93)

```python
# Dagster modules
from dagster_dbt import (
    DbtCliResource,
    load_assets_from_dbt_project
)

from dagster import (
    define_asset_job,
    job,
    op,
    Definitions,
    ScheduleDefinition,
    file_relative_path

)

DBT_PROJECT_PATH = file_relative_path( __file__, "../.." )
DBT_PROFILES_DIR = file_relative_path( __file__, "../../config" ) # default dir

dbt_assets = load_assets_from_dbt_project(
    project_dir = DBT_PROJECT_PATH,
    profiles_dir = DBT_PROFILES_DIR,
)

""" Resources """

my_dbt_resource = {
    "dbt": DbtCliClientResource(
        project_dir = DBT_PROJECT_PATH,
        profiles_dir = DBT_PROFILES_DIR,
    ),
}

""" Ops """

@op
def run_dbt_snapshots(dbt: DbtCliResource):
    dbt.cli(["snapshot"])

""" Jobs """

@job(resource_defs={"dbt": my_dbt_resource})
def dbt_snapshot_job():
    run_dbt_snapshots()

""" Definitions """

defs = Definitions(
    assets = [
        *dbt_assets,
    ],
    resources = {
        **my_dbt_resource,
    },
    schedules = [
        ScheduleDefinition(
            job = dbt_snapshot_job,
            cron_schedule="0 4 * * 0",
        )
    ],
)
```

**Comment by rexledesma** [comment, marked as answer]
I suggest that you follow the tutorial. You're mixing the old APIs (`load_assets_from_dbt_project`, `DbtCliClientResource`) with the new APIs (`DbtCliResource`).

To accomplish your use case, you can:
- Scaffold a Dagster project from a dbt project using `dagster-dbt project scaffold`
- In the `schedules.py` of your scaffolded project, you can create a job for only your snapshots.

```python
# schedules.py

from dagster_dbt import build_schedule_from_dbt_selection

from .assets import {{ dbt_assets_name }} # This will be created by the scaffold

schedules = [
    build_schedule_from_dbt_selection(
        [{{ dbt_assets_name }}],
        job_name="materialize_dbt_snapshots",
        cron_schedule="0 0 * * *",
        dbt_select="resource_type:snapshot,fqn:*",
    ),
]
```

References:
- [Tutorial for using Dagster and dbt Core](https://docs.dagster.io/integrations/dbt/using-dbt-with-dagster)
- [Scaffolding a Dagster project from a dbt project](https://docs.dagster.io/integrations/dbt/reference#scaffolding-a-dagster-project-from-a-dbt-project)
- [Scheduling a dbt job on Dagster](https://docs.dagster.io/integrations/dbt/reference#scheduling-jobs-that-contain-only-dbt-assets)

**Reply by freddyosprey** [reply]
@rexledesma thank you! I will give this a go
